app/driver.go

```diff
@@ -371,29 +372,134 @@ func getFlexvolumeDaemonSetSpec(image, flexvolumeDir string) *appsv1beta2.Daemon
 }
 
 func discoverFlexvolumeDir(kubeClient *clientset.Clientset) (dir string, err error) {
-	defer func() {
-		err = errors.Wrap(err, "cannot discover Flexvolume Dir")
-	}()
-	nodeName, err := util.GetRequiredEnv(types.EnvNodeName)
-	if err != nil {
-		return "", fmt.Errorf("Env %v wasn't set", types.EnvNodeName)
+	flexvolumeDir := ""
+
+	namespace := os.Getenv(types.EnvPodNamespace)
+	if namespace == "" {
+		return "", fmt.Errorf("Cannot detect pod namespace, environment variable %v is missing", types.EnvPodNamespace)
 	}
-	uri := fmt.Sprintf("/api/v1/proxy/nodes/%s/configz", nodeName)
-	rawConfigInBytes, err := kubeClient.Core().RESTClient().Get().RequestURI(uri).DoRaw()
-	if err != nil {
-		return "", errors.Wrapf(err, "cannot reach node config URI %v", uri)
+	serverPort := int32(8080)
+	objectMeta := metav1.ObjectMeta{
+		Name: "discover-flexvolume-dir",
 	}
-	jsonParsed, err := gabs.ParseJSON(rawConfigInBytes)
-	if err != nil {
-		return "", errors.Wrapf(err, "cannot parse json")
+
+	// start server to receive results
+	doneChan := make(chan struct{})
+	s := &http.Server{
+		Addr: fmt.Sprintf(":%d", serverPort),
 	}
-	value, ok := jsonParsed.Path("kubeletconfig.volumePluginDir").Data().(string)
-	if !ok {
-		logrus.Infof("cannot find volumePluginDir key in node config, assume it's default")
-		return DefaultFlexvolumeDir, nil
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Start a new web service for this? I think it's overkill.

I think you can just run a pod to completion then check it's log.

Reply:
I tend to agree, but haven't figured out how to access the logs. I'll dig into it more.